Added root /samples/README.md (#6998)
Azure/azure-sdk-for-js: samples/README.md

# Samples, Snippets, and How-To Guides

> Developers like to learn by looking at code, and so the Azure SDK comes with a myriad of code samples in the form of short code snippets, sample applications, and how-to guides. This document describes where to find all these resources.

Reviewer: ```suggestion
Developers like to learn by looking at code, and so the Azure SDK comes with many samples in the form of short code snippets, sample applications, and how-to guides. This document describes where to find all these resources.
```

Reviewer: "A myriad of" is technically correct, but might get us trolled vs. just "myriad" 😂. This simplifies the language a bit.

## Structure of the Repository
The Azure SDK repository is organized in the following folder structure, with the main sample locations highlighted using **bold** font.

- `/samples` (this folder)
  - `README.md` (this file)
- `/sdk` (folder containing sources for all SDK packages)
  - `/<service>` (e.g. storage)
    - `/<package>` (e.g. `@azure/storage-blob`)
      - **`README.md`** (package READMEs contain "hello world" code snippets)
      - **`/samples`**
        - **`/typescript`** (package-specific samples in TypeScript)
        - **`/javascript`** (package-specific samples in JavaScript using CommonJS)
      - `/src`
      - `/tests`

##  Getting Started (a.k.a. `Hello World`) Samples

> Each package folder contains a package-specific `README.md` file. Most of these `README` files contain `Hello World` code samples illustrating basic usage of the the APIs contained in the package. For example, you can find `Hello World` samples for the `@azure/storage-blob` package [here](https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/storage/storage-blob#examples).

Reviewer 2: Hm do we use the term `Hello World` in other docs/references? I think it is a good cultural touchstone for programmers, but not sure if we use it as a regular term or not

Author: .NET uses this term in its corresponding file. I'm not attached to it... we're considerably past the bare basics of the APIs in many of the readmes.

Personally -- I think that a lot of what's in our readmes is redundant with what's in the samples files, but that's a different question entirely. Maybe we could just call them "introductory samples" or something like that. Just "Getting Started Samples" would also work.

Reviewer 2: I have a small preference for "getting started code" or somesuch

## Package Samples and How-To Guides
Each package folder contains a subfolder called `samples` with additional code samples in the `typescript` and `javascript` subfolder. Corresponding files (i.e. `samples/typescript/src/helloWorld.ts` and `samples/javascript/helloWorld.js`) in the TypeScript and JavaScript directories are functionally equivalent. These samples may be viewed in GitHub or browsed on [the Sample Browser at docs.microsoft.com/samples](https://docs.microsoft.com/samples/browse/?products=azure&languages=typescript%2Cjavascript) by selecting the product and language combination that you wish to view samples for.

> ## Sample Applications

Reviewer 2: do we want to call out the bundling samples? 

https://github.com/Azure/azure-sdk-for-js/tree/master/samples/Bundling

It's the only non-package specific example we have right now, but I can imagine adding a couple more in the future to do generic things (e.g. how to test with our libs)

Author: yeah good point I'll make a quick change to this.

Reviewer 2: can you also add a link to the docs for this? https://github.com/Azure/azure-sdk-for-js/blob/master/documentation/Bundling.md

> Sometimes we want to illustrate how several APIs or even packages work together in a context of a more complete program. For these cases, we created sample applications that you can look at, download, compile, and execute. These application samples are located in other repositories and, like the package-specific samples, may be viewed on [https://docs.microsoft.com/samples/](https://docs.microsoft.com/samples/).

Reviewer: ```suggestion
Sometimes we want to illustrate how several APIs or even packages work together in a context of a more complete program. For these cases, we created sample applications. These application samples are located in other repositories and, like the package-specific samples, may be viewed on [https://docs.microsoft.com/samples/](https://docs.microsoft.com/samples/).
```

Not sure we need to enumerate the things you can do with programs :-P

# Samples, Snippets, and How-To Guides

Developers like to learn by looking at code, and so the Azure SDK comes with a myriad of code samples in the form of short code snippets, sample applications, and how-to guides. This document describes where to find all these resources.

## Structure of the Repository
The Azure SDK repository is organized in the following folder structure, with the main sample locations highlighted using **bold** font.

- `/samples` (this folder)
  - `README.md` (this file)
- `/sdk` (folder containing sources for all SDK packages)
  - `/<service>` (e.g. storage)
    - `/<package>` (e.g. `@azure/storage-blob`)

>       - **`README.md`** (package READMEs contain "hello world" samples)

Reviewer 3: samples -> code snippets?

```suggestion
      - **`README.md`** (package READMEs contain "hello world" code snippets)
```

>       - **`/samples`** (package-specific samples)

Reviewer 3: This made me think that `samples` folder has some samples and then the `typescript` and `javascript` folders had more samples.

Maybe skip the text in the parenthesis here?
